cou/steps/execute.py

```diff
@@ -59,7 +60,24 @@ async def _run_step(step: BaseStep, prompt: bool, overwrite_progress: bool = Fal
         logger.debug("running all sub-steps of %s step sequentially", step)
         for sub_step in step.sub_steps:
             logger.debug("running sub-step %s of %s step", sub_step, step)
-            await apply_step(sub_step, prompt, overwrite_substeps_progress)
+            try:
+                await apply_step(sub_step, prompt, overwrite_substeps_progress)
+            except HaltUpgradeExecution:
```

Comment:
IIUC based on the execution logic here, if instance check on vm failed, none of the steps in the same level will be executed, including `_get_enable_scheduler_step()` even though it doesn't have the `dependent` parameter. I don't think this is the correct behavior.

Reply:
The step raining `HaltUpgradeExecution` will handle the enabling itself. The dependent is only to visually improve UX.
With saying `progress_indicator.fail()` you will see something like this:
```bash
# pritning plan
...
Upgrade plan for unit: nova-compute/2
    Disable nova-compute scheduler from unit: 'nova-compute/2'.
    Check if unit nova-compute/2 has no VMs running before upgrading.
    ├── Pause the unit: 'nova-compute/2'.
    ├── Upgrade the unit: 'nova-compute/2'.
    ├── Resume the unit: 'nova-compute/2'.
    Enable nova-compute scheduler from unit: 'nova-compute/2'.
...
# executing plan
Upgrade plan for unit: nova-compute/1 ✔
Upgrade plan for unit: nova-compute/2 ✖
Upgrade plan for unit: nova-compute/3 ...
...
```

Reply:
Yes, but in the log, it will say that `Enable nova-compute scheduler from unit: 'nova-compute/2'.` step is skipped, right? That's is giving the user a false sense that nova-compute scheduler is still disabled.

Reply:
Wouldn't it be better if we don't enable the scheduler in "Check if unit nova-compute/2 has no VMs running before upgrading." but actually run  `Enable nova-compute scheduler from unit: 'nova-compute/2'` step?

Reply:
I guess we'll need to get a little dipper into this in another PR. For this one I'm removing the independent steps like `Enable nova-compute scheduler from unit: 'nova-compute/2'.` from the warning log so the user get less confuse about if the scheduler is enable or not.

I still think that it would be an easier implementation if the "dependency" was  as substeps of the step that can Halt. We can talk about this later

Reply:
I also think it would be better expressing dependencies with one more level of sub-steps. The overall logic would require more thorough design, especially considering the problem I pointed out in the other comment. 

For this PR, improving the warning message is enough.